build.gradle

```diff
@@ -32,6 +34,54 @@ compileJava {
 	options.release = 11
 }
 
+javadoc {
+	/* We don't actually need the full javadoc for anything, so just hijack this task for
+	 * building the client javadocs. If we ever need the full javadocs make a new task for the
+	 * client java docs
+	 */
+	/* TODO DOCS currently java-common and auth links don't work. ROI for fixing them is low.
+	 *           Ideally, in the future make github actions publish docs on release and ref them
+	 *           here.
+	 *           https://github.com/marketplace/actions/deploy-publish-javadoc
+	 *           https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-gradle-registry
+	 */
+	/* TODO DOCS the current sdk documenation looks like invalid html to javadoc
+	 *           need to go through and remove
+	 *           also some spots with < / > that need to be wrapped with {@code } in internal code
+	 */
+	failOnError = false
+	options {
+		links "https://docs.oracle.com/en/java/javase/11/docs/api/"
+		links "https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-core/2.9.9/"
+		links "https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.9.9/"
+	}
+	include "**/workspace/*.java"
+	exclude "**/workspace/WorkspaceServer.java"
+	include "**/common/service/Tuple*.java"
```

Comment:
@ialarmedalien I know you laboriously copied over all the options from ant in your PR, but frankly I have no idea why they were there in the first place or if they're needed. The docs without them look fine to me so in the interests of simplicity I left them out. If we find we need them in the future it's easy enough to put them back in

Reply:
Not sure why you didn't just reuse the code that I'd already written, esp since I laboriously tested it to ensure that it generated the same stuff. Also IMO it would have made more sense to start by building the jars, then the wars, then the docs, rather than going at things the other way around.

Reply:
I need the docs for the war and the shadow jar, which is why I'm going them first